routers/api/v1/repo/issue_reaction.go

```diff
@@ -258,8 +262,7 @@ func GetIssueReactions(ctx *context.APIContext) {
 		return
 	}
 
-	if !ctx.Repo.CanRead(models.UnitTypeIssues) && !ctx.User.IsAdmin {
-		ctx.Error(http.StatusForbidden, "GetIssueReactions", errors.New("no permission to get reactions"))
+	if !utils.HasUserAccessToIssues(ctx) {
```

Comment:
Hasn't this test been checked by

https://github.com/6543-forks/gitea/blob/a32425f070c67d56a894b1573f95f3e8ee9b17f8/routers/api/v1/api.go#L712

Reply:
`mustEnableIssuesOrPulls`:

https://github.com/6543-forks/gitea/blob/a32425f070c67d56a894b1573f95f3e8ee9b17f8/routers/api/v1/api.go#L456

only check if Issues are enabled - as far as i can tell

Reply:
Hmm but what about the issues endpoints above these changed reaction endpoints?

Reply:
no endpoints changed ?!? - just allow noneLogedin API requests to public repos @zeripath

Reply:
Remember this is checked too:

https://github.com/go-gitea/gitea/blob/aee1133e505a26a81ffad32c029ea093dc8feb92/routers/api/v1/api.go#L804

Either the reactions need these additional checks - in which case so do the issue comments endpoints, or this is just repeating work.

The issue comments and their reactions should have the same the permissions checks.